src/Development/IDE/Core/Debouncer.hs

```diff
@@ -39,6 +39,10 @@ newAsyncDebouncer = Debouncer . asyncRegisterEvent <$> newVar Map.empty
 -- Events are run unmasked so it is up to the user of `registerEvent`
 -- to mask if required.
 asyncRegisterEvent :: (Eq k, Hashable k) => Var (HashMap k (Async ())) -> Seconds -> k -> IO () -> IO ()
+asyncRegisterEvent d 0 k fire = modifyVar_ d $ \m -> mask_ $ do
+    whenJust (Map.lookup k m) cancel
+    fire
```

Comment:
Oops, forgot to unmask here

Reply:
Should we just disable the debouncer in the tests?

Reply:
But this is a real issue. If we start disabling things for test, we end up testing the wrong thing.

Reply:
Makes sense :+1:

Reply:
fixed